From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Peter Marshall <pm@petermarshall.ca>
Date: Thu, 30 Jul 2026 16:03:35 -0400
Subject: [PATCH] dt-bindings: media: i2c: st,vd55g: rename and add VD55G0
 variant

Document the STMicroelectronics VD55G0 camera sensor. Rename the device
tree bindings to represent a generic VD55G family sensor.

Signed-off-by: Peter Marshall <pm@petermarshall.ca>

Origin: petm5 (Peter Marshall), linux-surface/kernel PR 169 / linux-media series 1169036 (v2, 2026-09-18)
Upstream-Status: Submitted (v2); ST questions merging VD55G0 into the VD55G1 driver

---
 .../i2c/{st,vd55g1.yaml => st,vd55g.yaml}     | 30 +++++++++++++++----
 1 file changed, 24 insertions(+), 6 deletions(-)
 rename Documentation/devicetree/bindings/media/i2c/{st,vd55g1.yaml => st,vd55g.yaml} (76%)

diff --git a/Documentation/devicetree/bindings/media/i2c/st,vd55g1.yaml b/Documentation/devicetree/bindings/media/i2c/st,vd55g.yaml
similarity index 76%
rename from Documentation/devicetree/bindings/media/i2c/st,vd55g1.yaml
rename to Documentation/devicetree/bindings/media/i2c/st,vd55g.yaml
index 060ac6829..bd6feeb6c 100644
--- a/Documentation/devicetree/bindings/media/i2c/st,vd55g1.yaml
+++ b/Documentation/devicetree/bindings/media/i2c/st,vd55g.yaml
@@ -2,19 +2,21 @@
 # Copyright (c) 2025 STMicroelectronics SA.
 %YAML 1.2
 ---
-$id: http://devicetree.org/schemas/media/i2c/st,vd55g1.yaml#
+$id: http://devicetree.org/schemas/media/i2c/st,vd55g.yaml#
 $schema: http://devicetree.org/meta-schemas/core.yaml#
 
-title: STMicroelectronics VD55G1 Global Shutter Image Sensor
+title: STMicroelectronics VD55G Family Global Shutter Image Sensors
 
 maintainers:
   - Benjamin Mugnier <benjamin.mugnier@foss.st.com>
   - Sylvain Petinot <sylvain.petinot@foss.st.com>
+  - Peter Marshall <pm@petermarshall.ca>
 
 description: |-
- The STMicroelectronics VD55G1 is a global shutter image sensor with an active
- array size of 804H x 704V. It is programmable through I2C interface. The I2C
- address is fixed to 0x10.
+ The STMicroelectronics VD55G family (VD55G0, VD55G1, and VD65G4) consists of
+ global shutter image sensors. The VD55G0 has an array size of 644H x 604V.
+ The VD55G1 and VD65G4 have an array size of 804H x 704V. They are programmable
+ through I2C interface. The I2C address is fixed to 0x10.
 
  Image data is sent through MIPI CSI-2, which is configured as only 1 data
  lane. The sensor provides 4 GPIOS that can be used for external LED signal
@@ -23,13 +25,29 @@ description: |-
 allOf:
   - $ref: /schemas/media/video-interface-devices.yaml#
 
+  - if:
+      properties:
+        compatible:
+          contains:
+            const: st,vd55g0
+    then:
+      properties:
+        port:
+          properties:
+            endpoint:
+              properties:
+                link-frequencies:
+                  items:
+                    minimum: 375000000
+
 properties:
   compatible:
     enum:
+      - st,vd55g0
       - st,vd55g1
       - st,vd65g4
     description:
-      VD55G1 is the monochrome variant, while VD65G4 is the color one.
+      VD55G0 and VD55G1 are monochrome variants, while VD65G4 is the color variant.
 
   reg:
     maxItems: 1
-- 
2.55.0

